cns/restserver/homeazmonitor.go

```diff
@@ -165,8 +167,12 @@ func (h *HomeAzMonitor) update(code types.ResponseCode, msg string, homeAzRespon
 		},
 		HomeAzResponse: homeAzResponse,
 	}
-	logger.Printf("[HomeAzMonitor] updating home az cache value: %+v", resp)
-	h.updateCacheValue(resp)
+
+	// log the response and update the cache if it doesn't match with the current cached value
+	if !cmp.Equal(h.readCacheValue(), resp) {
```

Comment:
this looks like a value type (struct of value types), and could be compared with `==`.
https://go.dev/ref/spec#Comparison_operators
> Struct types are comparable if all their field types are comparable. Two struct values are equal if their corresponding non-[blank](https://go.dev/ref/spec#Blank_identifier) field values are equal. The fields are compared in source order, and comparison stops as soon as two field values differ (or all fields have been compared).